active/NumberBox/NumberBox.md

```diff
@@ -0,0 +1,184 @@
+the submission process.  When it's complete
+it will be incorporated into the public documentation at
+docs.microsoft.com (http://docs.microsoft.com/uwp/toolkits/winui/).
+Hopefully we'll be able to copy it mostly verbatim.
+So the second audience is everyone that reads there to learn how
+and why to use this API. -->
+
+## Background
+<!-- Use this section to provide background context for the new API(s) 
+in this spec. -->
+
+<!-- This section and the appendix are the only sections that likely
+do not get copied to docs.microsoft.com; they're just an aid to reading this spec. -->
+
+<!-- If you're modifying an existing API, included a link here to the
+existing page(s) -->
+
+<!-- For example, this section is a place to explain why you're adding this API rather than
+modifying an existing API. -->
+
+<!-- For example, this is a place to provide a brief explanation of some dependent
+area, just explanation enough to understand this new API, rather than telling
+the reader "go read 100 pages of background information posted at ...". -->
+
+
+## Description
+<!-- Use this section to provide a brief description of the feature.
+For an example, see the introduction to the PasswordBox control 
+(http://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box). -->
+
+Number box is a numerics only text control with support for input formatting, validation, basic calculation, and increment stepping.
+
+**Important APIs:** [NumberBox class](https://docs.microsoft.com/en-us/uwp/api/microsoft.ui.xaml.controls.numberbox) [TODO - generated with MSDN documentation on publication.]
+
+## Is this the right control? 
+
+Use a **NumberBox** control to receive and display formatted numeric values such as currency, IP addresses, and telephone numbers. A **NumberBox** can also be enabled with basic calculator support to compute multiplication, division, addition, and subtraction across parenthetical order with standard operator precedence.
+
+## Examples
+<!-- Use this section to explain the features of the API, showing
+example code with each description. The general format is: 
+  feature explanation,
+  example code
+  feature explanation,
+  example code
+  etc.-->
+  
+<!-- Code samples should be in C# and/or C++/WinRT -->
+
+<!-- As an example of this section, see the Examples section for the PasswordBox control 
+(https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). -->
+
+### Create a NumberBox
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="simple NumberBox"/>
+```
+
+![](images/numberbox-basic.png)
+
+### Add placeholder text
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="simple NumberBox" Header="Enter equation:" PlaceholderText="A + B * C" />
+```
+
+![](images/numberbox-placeholder-text.png)
+
+### Enable calculation support
+
+XAML
+```XAML
+<NumberBox AutomationProperties.Name="NumberBox for simple calculations" 
+    Header="Enter equation:" 
```

Comment:
Like Calculate, text is aligned to right side. I prefer to set alignment to right if AcceptsCalculation is true.
Also can we preview the output. For example, if you type '1+2', there is gray text followed with '=3'

Reply:
I spent a lot of time thinking about whether content inside the NumberBox should be aligned differently to a TextBox, as I nearly proposed this myself,  but came to the conclusion that it shouldn't. I considered what varying needs of different apps are and what Excel does (it switches internal alignment based on what it infers the content type of a cell to be.) 

I keep going back and forth on this but without hard data (I'm not sure this is even possible to obtain) for making a choice one way or another there will always be good arguments for both options. I'm also aware preference will be influenced by past personal experiences and the current needs of any developer making the decision.

My inclination is to keep the `HorizontalContentAlignment` the same as the parent control but consider changing it if during the preview the feedback from real world use of the control is that developers have to change it more than they don't.

Reply:
I like the idea of calculation preview assuming it doesn't introduce any unacceptable overhead! It could also be used to pre-validate correct entry for the user. These are things to consider. As far as text alignment goes, @mrlacey, we will have the pre-release period to try it out and collect feedback. My inclination would be the same as yours which is to keep HorizontalContentAlignment the same as the parent control unless we receive feedback otherwise.

Reply:
A calculation preview could be displayed as a tooltip that appears above the text field perhaps?

![image](https://user-images.githubusercontent.com/7389110/58842381-cff0e280-8665-11e9-8d80-8fd509e7ce75.png)

Reply:
These ideas are so cool. I'm going to add an open question bullet here and on the proposal for this "preview calculations" concept.

Reply:
Regarding the greyed out text example - I worry that this may encourage the user to fill in the values that appear, rather than act as a hint as to what the final result will be.  It looks just like the PlaceholderText.

What about using the AccentColor and different font weight to make it stand out
![image](https://user-images.githubusercontent.com/7389110/58920708-5b807700-872b-11e9-9924-21a7b7d37e68.png)

Reply:
I have the image updated here and on the Proposal. I think you are right that grayed out text could look like a prompt, which if followed, would result in an error due to the "=".

Reply:
Definitely prefer the in-box preview placement compared to the thhe tooltip placement. Since the actual calculation enetred so far and the preview result are directly related to each other they also have to be visually as close to each other as possible. In the above tooltip version, that's not given (Entering calculations on the left, then have to look above the input field all to the right.) In-box placement looks about right; have some padding betwenn the input and the review value as shown above and we should be all set.

If result preview is to be implemented, we should also think about adding a new property then, like
`public bool ShowResultPreview { get; set; }`
so it can be turned on/off by users.